### PR TITLE
Move sync from model to task

### PR DESCRIPTION
**Pull request by asmacdo**
closes #3260
https://pulp.plan.io/issues/3260

Depends on https://github.com/pulp/pulp/pull/3259

### CODE REVIEW

**Comment by jortel** [comment]
LGTM

File: pulp_file/app/tasks.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+    Raises:
+        ValueError: When feed_url is empty.
+    """
+    importer = file_models.FileImporter.objects.get(pk=importer_pk)
+
+    if not importer.feed_url:
+        raise ValueError(_("An importer must have a 'feed_url' attribute to sync."))
+
+    new_version = models.RepositoryVersion(repository=importer.repository)
```
Comment (dkliban) [inline pulp_file/app/tasks.py]: This needs to occur in a transaction together with adding to task.created_resources

Comment (asmacdo) [reply pulp_file/app/tasks.py]: I implemented this change, but something was broken. To get this PR merged, I've created another story for that work.
https://pulp.plan.io/issues/3274

**Comment by asmacdo** [comment]
https://github.com/pulp/pulp/pull/3261 is required to GET and serialize the Task with CreatedResource

**Pull request merged**